packages/core/http/core-http-server-internal/src/http_server.ts

```diff
@@ -526,6 +527,10 @@ export class HttpServer {
       xsrfRequired: route.options.xsrfRequired ?? !isSafeMethod(route.method),
       access: route.options.access ?? (route.path.startsWith('/internal') ? 'internal' : 'public'),
     };
+    // Log HTTP API target consumer. Warning: may log sensitive information on paths including secrets
```

Comment:
I added a warning to remind ourselves (and API owners) where debug logs could cause issues.

Reply:
IIUC, something like `POST /api/my-secret/is/:mySecret` would log `... for path [/api/my-secret/is/:mySecret]`, and not the actual value in the secret when the user performs the request. Am I missing anything?

Reply:
You're not missing anything, the warning is to remind folks that anything in the path will log as is and to think very carefully about what they are fine with exposing is.

Reply:
Sorry... I'm a bit slow today... 😞 

AFAIK, this comment is only read by Kibana Core devs. I don't get how this comment would warn a plugin developer to design their API so that they don't **register** an HTTP route like `POST /api/my/plugin/${config.secret}/:id`. Bear in mind we are logging the registered path, not the final path that is requested with the variables replaced by the values (`:id` in the example)

Reply:
@afharo I believe what you are describing has already been introduced with https://github.com/elastic/kibana/blob/8fdcf0956760a5fca6abaf73238c5f19cebeeaac/packages/core/http/core-http-server-internal/src/http_server.ts#L519

Reply:
Not a reason to review the decision, just saying it's not just being added with this PR :)

Reply:
I don't want to block this PR 😇. These are nits rather than anything. Also, I noticed @TinaHeiligers removed the comment.

> @afharo I believe what you are describing has already been introduced with
> 
> https://github.com/elastic/kibana/blob/8fdcf0956760a5fca6abaf73238c5f19cebeeaac/packages/core/http/core-http-server-internal/src/http_server.ts#L519

IIUC, I'm describing both lines. This PR adds a new entry in the logs that _rephrases and adds info_ to the already existing entry. The new log entry is also logging the registered path, not the accessed path.